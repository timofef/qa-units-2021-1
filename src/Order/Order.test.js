/* eslint-disable import/first */
jest.mock("../utils/getDate");

import {getDate} from "../utils/getDate";
import React from 'react'
import Order from "./Order";
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Order.js', () => {
  beforeEach(() => {
    getDate.mockReturnValue("4 августа, чт, 2021 год");
  });

  afterEach(() => {
    jest.resetAllMocks()
  });

  it('render default', () => {
    const wrapper = shallow(<Order/>);

    expect(wrapper.isEmptyRender()).toBeTruthy();
  });

  it('render with data', () => {
    const wrapper = shallow(<Order order={{shop: "some", date: 1}}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('render with data and items', () => {
    const wrapper = shallow(<Order order={{shop: "some", date: 1, items: [1, 2, 3, 4]}}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('render with no date', () => {
    const wrapper = shallow(<Order order={{shop: "some"}}/>);

    expect(wrapper.isEmptyRender()).toBeTruthy();
  });

  it('render with no shop', () => {
    const wrapper = shallow(<Order order={{date: 1}}/>);

    expect(wrapper.isEmptyRender()).toBeTruthy();
  });

  it('render without shop and date', () => {
    const wrapper = shallow(<Order order={{}}/>);

    expect(wrapper.isEmptyRender()).toBeTruthy();
  });

  it('check if getDate is called', () => {
    const testData = {shop: "some", date: 1};
    const wrapper = shallow(<Order order={testData}/>);

    expect(getDate).toHaveBeenCalledWith(testData.date);
  });
});

