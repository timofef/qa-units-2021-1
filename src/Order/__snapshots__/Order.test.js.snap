// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js render default 1`] = `""`;

exports[`Order.js render with data 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      some
    </span>
    <span
      id="date"
    >
      4 августа, чт, 2021 год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order.js render with data and items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      some
    </span>
    <span
      id="date"
    >
      4 августа, чт, 2021 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      1
    </div>
    <div
      className="Order-item"
      key="1"
    >
      2
    </div>
    <div
      className="Order-item"
      key="2"
    >
      3
    </div>
    <div
      className="Order-item"
      key="3"
    >
      4
    </div>
  </div>
</div>
`;

exports[`Order.js render with no date 1`] = `""`;

exports[`Order.js render with no shop 1`] = `""`;

exports[`Order.js render without shop and date 1`] = `""`;
